File: LeetCode/Java/src/problems/p0058_LengthOfLastWord/P0058.java
package problems.p0058_LengthOfLastWord;

public class P0058 {
    public int lengthOfLastWord(String s) {
        /*
         * Solution v1: Brute Force
         *
         * Runtime: 0 ms @ (beats) 100%
         * Memory Usage: 38.4 MB @ (beats) 33.03%
         */
        // if (s.length() == 1) {
        //     if (!s.equals(" ")) {
        //         return s.length();
        //     } else {
        //         return 0;
        //     }
        // }
        // s = s.trim();
        // int ans = 0, i = s.length() - 1;
        // while (i >= 0) {
        //     if (s.charAt(i) == ' ') {
        //         return ans;
        //     }
        //     ans++;
        //     i--;
        // }
        // return ans;

        /*
         * Solution v2: Use String class
         *
         * Runtime: ms @ (beats) %
         * Memory Usage: MB @ (beats) %
         */
        // s = s.trim();
        // return s.length() - s.lastIndexOf(" ") - 1;

        /*
         * (learn from Internet)
         * Solution v3: Convert to string array
         */
        String[] split = s.split(" ");
        return split[split.length - 1].length();
    }
}

File: LeetCode/Python/solutions/p0058.py
class Solution:
    def lengthOfLastWord(self, s: str) -> int:
        #
        # Solution v1: Brute Force
        #
        # Runtime: 32 ms @ (beats) 59.70%
        # Memory Usage: 14.3 MB @ (beats) 36.79%
        #
        s = s.rstrip()
        i = len(s) - 1
        ans = 0
        while i >= 0:
            if s[i] != " ":
                ans += 1
                i -= 1
            else:
                break
        return ans
